### PR TITLE
add unit test for ServerChannels

### PR DESCRIPTION
**Pull request by sullis**

### CODE REVIEW

File: pinot-core/src/test/java/org/apache/pinot/core/transport/ServerChannelsTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+  @DataProvider
+  public Object[][] parameters() {
+    return new Object[][]{
+        new Object[]{true}, new Object[]{false},
+    };
+  }
+
+  @BeforeClass
+  public void beforeClass()
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/core/transport/ServerChannelsTest.java:50]: (convention) We usually name it `setUp()`

File: pinot-core/src/test/java/org/apache/pinot/core/transport/ServerChannelsTest.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      throws Exception {
+    _dummyServer = HttpServer.create();
+    _dummyServer.bind(new InetSocketAddress("localhost", 0), 0);
+    _dummyServer.start();
+  }
+
+  @AfterClass
+  public void afterClass()
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/core/transport/ServerChannelsTest.java:58]: (convention) We usually name it `tearDown()`

File: pinot-core/src/test/java/org/apache/pinot/core/transport/ServerChannelsTest.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      throws Exception {
+    if (_dummyServer != null) {
+      _dummyServer.stop(0);
+    }
+  }
+
+  @Test(dataProvider = "parameters")
+  public void testConnect(final boolean nativeTransportEnabled)
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/core/transport/ServerChannelsTest.java:66]: (convention) We don't usually put `final` for local variables

File: pinot-core/src/test/java/org/apache/pinot/core/transport/ServerChannelsTest.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      throws Exception {
+    PinotMetricUtils.init(new PinotConfiguration());
+    PinotMetricsRegistry registry = PinotMetricUtils.getPinotMetricsRegistry();
+    BrokerMetrics brokerMetrics = new BrokerMetrics(registry);
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/core/transport/ServerChannelsTest.java:70]: We can mock the `BrokerMetrics`

**Review by Jackie-Jiang — APPROVED** [review]
LGTM with minor comments

File: pinot-core/src/test/java/org/apache/pinot/core/transport/ServerChannelsTest.java (hunk 4 above)
Comment (sullis) [reply pinot-core/src/test/java/org/apache/pinot/core/transport/ServerChannelsTest.java]: done

File: pinot-core/src/test/java/org/apache/pinot/core/transport/ServerChannelsTest.java (hunk 3 above)
Comment (sullis) [reply pinot-core/src/test/java/org/apache/pinot/core/transport/ServerChannelsTest.java]: done

File: pinot-core/src/test/java/org/apache/pinot/core/transport/ServerChannelsTest.java (hunk 2 above)
Comment (sullis) [reply pinot-core/src/test/java/org/apache/pinot/core/transport/ServerChannelsTest.java]: done

File: pinot-core/src/test/java/org/apache/pinot/core/transport/ServerChannelsTest.java (hunk 1 above)
Comment (sullis) [reply pinot-core/src/test/java/org/apache/pinot/core/transport/ServerChannelsTest.java]: done

**Pull request merged**